format-check.sh

```diff
@@ -10,7 +10,7 @@ if NOT type $CLANG_FORMAT 2> /dev/null ; then
 fi
 
 FAIL=0
-SOURCE_FILES=`find source include tests -type f \( -name '*.h' -o -name '*.cpp' \) -not -name "cJSON.h" -not -name "cJSON.cpp"`
+SOURCE_FILES=`find source include tests samples canary -type f \( -name '*.h' -o -name '*.cpp' \) -not -name "cJSON.h" -not -name "cJSON.cpp"`
```

Comment:
I was going to ask how the formatting was getting past clang-format, NOW I SEE THE VIOLENCE INHERENT IN THE SYSTEM

Reply:
It shouldn't be though, right, since it's being added here?  (Although that samples shouldn't be there, I'll take that out.)